Gemfile

```diff
@@ -17,18 +17,27 @@ group :development, :test do
   gem 'pry-byebug' unless ENV['CI']
 end
 
-file = File.expand_path("Gemfile", ENV['ENGINE_CART_DESTINATION'] || ENV['RAILS_ROOT'] || File.expand_path("../spec/internal", __FILE__))
+
+# BEGIN ENGINE_CART BLOCK
+# engine_cart: 0.8.0
+# engine_cart stanza: 0.8.0
+# the below comes from engine_cart, a gem used to test this Rails engine gem in the context of a Rails app.
+file = File.expand_path("Gemfile", ENV['ENGINE_CART_DESTINATION'] || ENV['RAILS_ROOT'] || File.expand_path(".internal_test_app", File.dirname(__FILE__)))
 if File.exists?(file)
-  puts "Loading #{file} ..." if $DEBUG # `ruby -d` or `bundle -v`
-  instance_eval File.read(file)
+  begin
+    eval_gemfile file
+  rescue Bundler::GemfileError => e
+    Bundler.ui.warn '[EngineCart] Skipping Rails application dependencies:'
+    Bundler.ui.warn e.message
+  end
 else
+  Bundler.ui.warn "[EngineCart] Unable to find test application dependencies in #{file}, using placeholder dependencies"
   gem 'rails', ENV['RAILS_VERSION'] if ENV['RAILS_VERSION']
-  if ENV['RAILS_VERSION'] and ENV['RAILS_VERSION'] !~ /^4.2/
-    gem 'sass-rails', "< 5.0"
-  else
+  if ENV['RAILS_VERSION'].nil? || ENV['RAILS_VERSION'] =~ /^4.2/
+    gem 'responders', "~> 2.0"
     gem 'sass-rails', ">= 5.0"
+  else
+    gem 'sass-rails', "< 5.0"
   end
-
-  extra_file = File.expand_path("../spec/test_app_templates/Gemfile.extra", __FILE__)
```

Comment:
It's handled here: https://github.com/cbeer/engine_cart/blob/9ab2a6271d6b95c406961be7e03ffb3a764780ef/lib/engine_cart/tasks/engine_cart.rake#L67